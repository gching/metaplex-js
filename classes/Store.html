<!doctype html>
<html class="default no-js">
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>Store | metaplex-js</title>
	<meta name="description" content="Documentation for metaplex-js">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="../assets/css/main.css">
	<script async src="../assets/js/search.js" id="search-script"></script>
</head>
<body>
<header>
	<div class="tsd-page-toolbar">
		<div class="container">
			<div class="table-wrap">
				<div class="table-cell" id="tsd-search" data-index="../assets/js/search.json" data-base="..">
					<div class="field">
						<label for="tsd-search-field" class="tsd-widget search no-caption">Search</label>
						<input id="tsd-search-field" type="text" />
					</div>
					<ul class="results">
						<li class="state loading">Preparing search index...</li>
						<li class="state failure">The search index is not available</li>
					</ul>
					<a href="../index.html" class="title">metaplex-js</a>
				</div>
				<div class="table-cell" id="tsd-widgets">
					<div id="tsd-filter">
						<a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a>
						<div class="tsd-filter-group">
							<div class="tsd-select" id="tsd-filter-visibility">
								<span class="tsd-select-label">All</span>
								<ul class="tsd-select-list">
									<li data-value="public">Public</li>
									<li data-value="protected">Public/Protected</li>
									<li data-value="private" class="selected">All</li>
								</ul>
							</div>
							<input type="checkbox" id="tsd-filter-inherited" checked />
							<label class="tsd-widget" for="tsd-filter-inherited">Inherited</label>
							<input type="checkbox" id="tsd-filter-externals" checked />
							<label class="tsd-widget" for="tsd-filter-externals">Externals</label>
						</div>
					</div>
					<a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a>
				</div>
			</div>
		</div>
	</div>
	<div class="tsd-page-title">
		<div class="container">
			<ul class="tsd-breadcrumb">
				<li>
					<a href="../modules.html">metaplex-js</a>
				</li>
				<li>
					<a href="Store.html">Store</a>
				</li>
			</ul>
			<h1>Class Store</h1>
		</div>
	</div>
</header>
<div class="container container-main">
	<div class="row">
		<div class="col-8 col-content">
			<section class="tsd-panel tsd-hierarchy">
				<h3>Hierarchy</h3>
				<ul class="tsd-hierarchy">
					<li>
						<span class="target">Store</span>
					</li>
				</ul>
			</section>
			<section class="tsd-panel-group tsd-index-group">
				<h2>Index</h2>
				<section class="tsd-panel tsd-index-panel">
					<div class="tsd-index-content">
						<section class="tsd-index-section ">
							<h3>Constructors</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-constructor tsd-parent-kind-class"><a href="Store.html#constructor" class="tsd-kind-icon">constructor</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Properties</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Store.html#accountPublicKey" class="tsd-kind-icon">account<wbr>Public<wbr>Key</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Store.html#auctionProgram" class="tsd-kind-icon">auction<wbr>Program</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Store.html#key" class="tsd-kind-icon">key</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Store.html#public" class="tsd-kind-icon">public</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Store.html#tokenMetadataProgram" class="tsd-kind-icon">token<wbr>Metadata<wbr>Program</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Store.html#tokenProgram" class="tsd-kind-icon">token<wbr>Program</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Store.html#tokenVaultProgram" class="tsd-kind-icon">token<wbr>Vault<wbr>Program</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><a href="Store.html#SCHEMA" class="tsd-kind-icon">SCHEMA</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Accessors</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-get-signature tsd-parent-kind-class"><a href="Store.html#isPublic" class="tsd-kind-icon">is<wbr>Public</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Methods</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-method tsd-parent-kind-class tsd-is-static"><a href="Store.html#decode" class="tsd-kind-icon">decode</a></li>
							</ul>
						</section>
					</div>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Constructors</h2>
				<section class="tsd-panel tsd-member tsd-kind-constructor tsd-parent-kind-class">
					<a name="constructor" class="tsd-anchor"></a>
					<h3>constructor</h3>
					<ul class="tsd-signatures tsd-kind-constructor tsd-parent-kind-class">
						<li class="tsd-signature tsd-kind-icon">new <wbr>Store<span class="tsd-signature-symbol">(</span>args<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Omit</span><span class="tsd-signature-symbol">&lt;</span><a href="Store.html" class="tsd-signature-type" data-tsd-kind="Class">Store</a><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">&quot;key&quot;</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">&quot;isPublic&quot;</span><span class="tsd-signature-symbol">&gt;</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><a href="Store.html" class="tsd-signature-type" data-tsd-kind="Class">Store</a></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/Store.ts#L48">models/programs/metaplex/Store.ts:48</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>args: <span class="tsd-signature-type">Omit</span><span class="tsd-signature-symbol">&lt;</span><a href="Store.html" class="tsd-signature-type" data-tsd-kind="Class">Store</a><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">&quot;key&quot;</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">&quot;isPublic&quot;</span><span class="tsd-signature-symbol">&gt;</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <a href="Store.html" class="tsd-signature-type" data-tsd-kind="Class">Store</a></h4>
						</li>
					</ul>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Properties</h2>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="accountPublicKey" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagOptional">Optional</span> account<wbr>Public<wbr>Key</h3>
					<div class="tsd-signature tsd-kind-icon">account<wbr>Public<wbr>Key<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/Store.ts#L36">models/programs/metaplex/Store.ts:36</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="auctionProgram" class="tsd-anchor"></a>
					<h3>auction<wbr>Program</h3>
					<div class="tsd-signature tsd-kind-icon">auction<wbr>Program<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/Store.ts#L39">models/programs/metaplex/Store.ts:39</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="key" class="tsd-anchor"></a>
					<h3>key</h3>
					<div class="tsd-signature tsd-kind-icon">key<span class="tsd-signature-symbol">:</span> <a href="../enums/MetaplexKey.html" class="tsd-signature-type" data-tsd-kind="Enumeration">MetaplexKey</a><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/Store.ts#L37">models/programs/metaplex/Store.ts:37</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="public" class="tsd-anchor"></a>
					<h3>public</h3>
					<div class="tsd-signature tsd-kind-icon">public<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">0</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">1</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/Store.ts#L38">models/programs/metaplex/Store.ts:38</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="tokenMetadataProgram" class="tsd-anchor"></a>
					<h3>token<wbr>Metadata<wbr>Program</h3>
					<div class="tsd-signature tsd-kind-icon">token<wbr>Metadata<wbr>Program<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/Store.ts#L41">models/programs/metaplex/Store.ts:41</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="tokenProgram" class="tsd-anchor"></a>
					<h3>token<wbr>Program</h3>
					<div class="tsd-signature tsd-kind-icon">token<wbr>Program<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/Store.ts#L42">models/programs/metaplex/Store.ts:42</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="tokenVaultProgram" class="tsd-anchor"></a>
					<h3>token<wbr>Vault<wbr>Program</h3>
					<div class="tsd-signature tsd-kind-icon">token<wbr>Vault<wbr>Program<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/Store.ts#L40">models/programs/metaplex/Store.ts:40</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class tsd-is-static">
					<a name="SCHEMA" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagStatic">Static</span> SCHEMA</h3>
					<div class="tsd-signature tsd-kind-icon">SCHEMA<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">BorshSchema</span><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/Store.ts#L19">models/programs/metaplex/Store.ts:19</a></li>
						</ul>
					</aside>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Accessors</h2>
				<section class="tsd-panel tsd-member tsd-kind-get-signature tsd-parent-kind-class">
					<a name="isPublic" class="tsd-anchor"></a>
					<h3>is<wbr>Public</h3>
					<ul class="tsd-signatures tsd-kind-get-signature tsd-parent-kind-class">
						<li class="tsd-signature tsd-kind-icon"><span class="tsd-signature-symbol">get</span> isPublic<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">boolean</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/Store.ts#L44">models/programs/metaplex/Store.ts:44</a></li>
								</ul>
							</aside>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">boolean</span></h4>
						</li>
					</ul>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Methods</h2>
				<section class="tsd-panel tsd-member tsd-kind-method tsd-parent-kind-class tsd-is-static">
					<a name="decode" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagStatic">Static</span> decode</h3>
					<ul class="tsd-signatures tsd-kind-method tsd-parent-kind-class tsd-is-static">
						<li class="tsd-signature tsd-kind-icon">decode<span class="tsd-signature-symbol">(</span>accountWithPublicKey<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">AccountWithPublicKey</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><a href="Store.html" class="tsd-signature-type" data-tsd-kind="Class">Store</a></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/Store.ts#L52">models/programs/metaplex/Store.ts:52</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>accountWithPublicKey: <span class="tsd-signature-type">AccountWithPublicKey</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <a href="Store.html" class="tsd-signature-type" data-tsd-kind="Class">Store</a></h4>
						</li>
					</ul>
				</section>
			</section>
		</div>
		<div class="col-4 col-menu menu-sticky-wrap menu-highlight">
			<nav class="tsd-navigation primary">
				<ul>
					<li class=" ">
						<a href="../modules.html">Exports</a>
					</li>
				</ul>
			</nav>
			<nav class="tsd-navigation secondary menu-sticky">
				<ul class="before-current">
				</ul>
				<ul class="current">
					<li class="current tsd-kind-class">
						<a href="Store.html" class="tsd-kind-icon">Store</a>
						<ul>
							<li class=" tsd-kind-constructor tsd-parent-kind-class">
								<a href="Store.html#constructor" class="tsd-kind-icon">constructor</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Store.html#accountPublicKey" class="tsd-kind-icon">account<wbr>Public<wbr>Key</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Store.html#auctionProgram" class="tsd-kind-icon">auction<wbr>Program</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Store.html#key" class="tsd-kind-icon">key</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Store.html#public" class="tsd-kind-icon">public</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Store.html#tokenMetadataProgram" class="tsd-kind-icon">token<wbr>Metadata<wbr>Program</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Store.html#tokenProgram" class="tsd-kind-icon">token<wbr>Program</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Store.html#tokenVaultProgram" class="tsd-kind-icon">token<wbr>Vault<wbr>Program</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class tsd-is-static">
								<a href="Store.html#SCHEMA" class="tsd-kind-icon">SCHEMA</a>
							</li>
							<li class=" tsd-kind-get-signature tsd-parent-kind-class">
								<a href="Store.html#isPublic" class="tsd-kind-icon">is<wbr>Public</a>
							</li>
							<li class=" tsd-kind-method tsd-parent-kind-class tsd-is-static">
								<a href="Store.html#decode" class="tsd-kind-icon">decode</a>
							</li>
						</ul>
					</li>
				</ul>
				<ul class="after-current">
				</ul>
			</nav>
		</div>
	</div>
</div>
<footer class="with-border-bottom">
	<div class="container">
		<h2>Legend</h2>
		<div class="tsd-legend-group">
			<ul class="tsd-legend">
				<li class="tsd-kind-constructor tsd-parent-kind-class"><span class="tsd-kind-icon">Constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-class"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class"><span class="tsd-kind-icon">Method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-interface"><span class="tsd-kind-icon">Property</span></li>
			</ul>
		</div>
	</div>
</footer>
<div class="container tsd-generator">
	<p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p>
</div>
<div class="overlay"></div>
<script src="../assets/js/main.js"></script>
</body>
</html>