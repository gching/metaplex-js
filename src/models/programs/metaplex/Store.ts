import { PublicKey } from '@solana/web3.js';
import { deserializeUnchecked } from 'borsh';
import { BorshSchema } from '../../utils/types';
import { AccountWithPublicKey } from '../../../utils/privateTypes';

export enum MetaplexKey {
  Uninitialized = 0,
  OriginalAuthorityLookupV1 = 1,
  BidRedemptionTicketV1 = 2,
  StoreV1 = 3,
  WhitelistedCreatorV1 = 4,
  PayoutTicketV1 = 5,
  SafetyDepositValidationTicketV1 = 6,
  AuctionManagerV1 = 7,
  PrizeTrackingTicketV1 = 8,
}

export class Store {
  static SCHEMA: BorshSchema = new Map([
    [
      Store,
      {
        kind: 'struct',
        fields: [
          ['key', 'u8'],
          ['public', 'u8'],
          ['auctionProgram', 'pubkey'],
          ['tokenVaultProgram', 'pubkey'],
          ['tokenMetadataProgram', 'pubkey'],
          ['tokenProgram', 'pubkey'],
        ],
      },
    ],
  ]);

  accountPublicKey?: PublicKey;
  key: MetaplexKey = MetaplexKey.StoreV1;
  public: 0 | 1;
  auctionProgram: PublicKey;
  tokenVaultProgram: PublicKey;
  tokenMetadataProgram: PublicKey;
  tokenProgram: PublicKey;

  get isPublic(): boolean {
    return this.public === 1;
  }

  constructor(args: Omit<Store, 'key' | 'isPublic'>) {
    Object.assign(this, args);
  }

  static decode(accountWithPublicKey: AccountWithPublicKey): Store {
    const store = deserializeUnchecked(
      Store.SCHEMA,
      Store,
      accountWithPublicKey.account.data,
    ) as Store;

    store.accountPublicKey = accountWithPublicKey.pubkey;

    return store;
  }
}
