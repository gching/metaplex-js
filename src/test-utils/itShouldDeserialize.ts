import { PublicKey } from '@solana/web3.js';
import { serialize } from 'borsh';
import { BorshSchema } from '../models/utils/types';

export function itShouldDeserialize<E>(
  cls: {
    SCHEMA: BorshSchema;
    // eslint-disable-next-line @typescript-eslint/ban-types
    decode: Function;
  },
  received: E,
  accountPubKey = PublicKey.default,
): void {
  const data = Buffer.from(serialize(cls.SCHEMA, received));

  const deserialized = cls.decode({
    pubkey: accountPubKey,
    account: {
      data,
    },
  }) as E;

  expect(deserialized).toDeepEqual(received);
}
