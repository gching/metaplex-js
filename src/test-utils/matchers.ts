import { isEqualWith } from 'lodash';
import BN from 'bn.js';

declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace jest {
    interface Matchers<R> {
      toDeepEqual: <E = any>(expected: E) => R;
    }
  }
}

expect.extend({
  toDeepEqual<E = any>(received: E, expected: E) {
    const pass = isEqualWith(received, expected, (objValue, otherValue) => {
      if (BN.isBN(objValue) && BN.isBN(otherValue)) {
        return objValue.eq(otherValue);
      }
      return undefined;
    });

    if (pass) {
      return {
        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
        message: () => `expected ${received} not to deep equal ${expected}`,
        pass: true,
      };
    }

    return {
      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
      message: () => `expected ${received} to deep equal ${expected}`,
      pass: false,
    };
  },
});

export default undefined;
