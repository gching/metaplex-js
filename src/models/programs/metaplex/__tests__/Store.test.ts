import { PublicKey } from '@solana/web3.js';
import { itShouldDeserialize } from '../../../../test-utils/itShouldDeserialize';
import { Store } from '../Store';

describe('Store', () => {
  it('should instantiate with the correct values', () => {
    const inst = new Store({
      accountPublicKey: PublicKey.default,
      tokenVaultProgram: PublicKey.default,
      auctionProgram: PublicKey.default,
      tokenMetadataProgram: PublicKey.default,
      tokenProgram: PublicKey.default,
      public: 0,
    });

    expect(inst).toMatchSnapshot();
  });

  it('should deserialize with the correct values', () => {
    const inst = new Store({
      accountPublicKey: PublicKey.default,
      tokenVaultProgram: PublicKey.default,
      auctionProgram: PublicKey.default,
      tokenMetadataProgram: PublicKey.default,
      tokenProgram: PublicKey.default,
      public: 0,
    });

    inst.accountPublicKey = PublicKey.default;

    itShouldDeserialize(Store, inst);
  });
});
