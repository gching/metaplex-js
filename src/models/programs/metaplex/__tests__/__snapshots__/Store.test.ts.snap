// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store should instantiate with the correct values 1`] = `
Store {
  "accountPublicKey": PublicKey {
    "_bn": "00",
  },
  "auctionProgram": PublicKey {
    "_bn": "00",
  },
  "key": 3,
  "public": 0,
  "tokenMetadataProgram": PublicKey {
    "_bn": "00",
  },
  "tokenProgram": PublicKey {
    "_bn": "00",
  },
  "tokenVaultProgram": PublicKey {
    "_bn": "00",
  },
}
`;
